src/services/__tests__/messagesService.test.ts

```diff
@@ -888,3 +896,94 @@ describe("MessageService#upsertServicePreferences", () => {
     }
   );
 });
+
+describe("MessageService#upsertMessageStatus", () => {
+  const aMessageStatusChange: MessageStatusChange = {
+    change_type: Reading_Change_typeEnum.reading,
+    is_read: true
+  };
+
+  const aMessageStatus: MessageStatus = {
+    status: MessageStatusValueEnum.PROCESSED,
+    updated_at: new Date(),
+    version: 1
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should make the correct api call", async () => {
+    mockUpsertMessageStatus.mockImplementation(() => {
+      return t.success({
+        status: 200,
+        value: aMessageStatus
+      });
+    });
+
+    const service = new MessageService(
+      api,
+      {} as IPecServerClientFactoryInterface
+    );
+    const res = await service.upsertMessageStatus(
+      mockedUser.fiscal_code,
+      aValidMessageId as NonEmptyString,
+      aMessageStatusChange
+    );
+
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessJson",
+      value: aMessageStatus
+    });
+
+    expect(mockUpsertMessageStatus).toHaveBeenCalledWith({
+      fiscal_code: mockedUser.fiscal_code,
+      id: aValidMessageId,
+      body: aMessageStatusChange
+    });
+  });
+
+  it.each`
+    title                                                            | status_code | value   | expected_status_code | expected_kind                      | expected_detail
+    ${"return IResponseErrorInternal if status is 401"}              | ${401}      | ${null} | ${500}               | ${"IResponseErrorInternal"}        | ${"Internal server error: Underlying API fails with an unexpected 401"}
+    ${"return IResponseErrorNotFound if status is 404"}              | ${404}      | ${null} | ${404}               | ${"IResponseErrorNotFound"}        | ${"Not Found: Message not found"}
+    ${"return IResponseErrorTooManyRequests if status is 429"}       | ${429}      | ${null} | ${429}               | ${"IResponseErrorTooManyRequests"} | ${"Too many requests: "}
+    ${"return IResponseErrorInternal if status code is not in spec"} | ${418}      | ${null} | ${500}               | ${"IResponseErrorInternal"}        | ${"Internal server error: unhandled API response status [418]"}
+  `(
+    "should $title",
+    async ({
+      status_code,
+      value,
+      expected_status_code,
+      expected_kind,
+      expected_detail
```

Comment:
We should use camel case for typescript variables
```suggestion
      statusCode,
      value,
      expectedStatusCode,
      expectedKind,
      expectedDetail
```

Reply:
Thanks. Done in [1603360](https://github.com/pagopa/io-backend/pull/879/commits/1603360fd0792af5c2def72cf23c5eef58023106)